pkg/ipmi/blob_transfer.go

```diff
@@ -0,0 +1,416 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// OpenBMC IPMI Blob Protocol commands
+// This file declares functions that implement the generic blob transfer
+// interface detailed at https://github.com/openbmc/phosphor-ipmi-blobs
+// with IPMI as a transport layer.
+// See https://github.com/openbmc/google-ipmi-i2c for details on OEM
+// commands.
+
+package ipmi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math/rand"
+	"unsafe"
+)
+
+// CRCOption is an option for sending/receiving CRCs.
+type CRCOption string
+
+// SessionID is a unique identifier for an open blob.
+type SessionID uint16
+
+// BlobStats contains statistics for a given blob.
+type BlobStats struct {
+	state       uint16
+	size        uint32
+	metadataLen uint8
+	metadata    []uint8
+}
+
+const (
+	IPMI_MAX_PAYLOAD_SIZE = 256
+
+	_IPMI_GGL_NET_FN   = 46
+	_IPMI_GGL_LUN      = 0
+	_IPMI_GGL_BLOB_CMD = 128
+
+	OEN_LEN = 3
+	CRC_LEN = 2
+
+	// Blob transfer command codes
+	BMC_BLOB_CMD_CODE_GET_COUNT    = 0
+	BMC_BLOB_CMD_CODE_ENUMERATE    = 1
+	BMC_BLOB_CMD_CODE_OPEN         = 2
+	BMC_BLOB_CMD_CODE_READ         = 3
+	BMC_BLOB_CMD_CODE_WRITE        = 4
+	BMC_BLOB_CMD_CODE_COMMIT       = 5
+	BMC_BLOB_CMD_CODE_CLOSE        = 6
+	BMC_BLOB_CMD_CODE_DELETE       = 7
+	BMC_BLOB_CMD_CODE_STAT         = 8
+	BMC_BLOB_CMD_CODE_SESSION_STAT = 9
+
+	// Flags
+	BMC_BLOB_OPEN_FLAG_READ  = (1 << 0)
+	BMC_BLOB_OPEN_FLAG_WRITE = (1 << 1)
+	// Blob open: bit positions 2-7 are reserved for future protocol use.
+	// Bit positions 8-15 are available for blob-specific definitions.
+	BMC_BLOB_STATE_OPEN_R       = (1 << 0)
+	BMC_BLOB_STATE_OPEN_W       = (1 << 1)
+	BMC_BLOB_STATE_COMMITTING   = (1 << 2)
+	BMC_BLOB_STATE_COMMITTED    = (1 << 3)
+	BMC_BLOB_STATE_COMMIT_ERROR = (1 << 4)
+	// Blob state: bit positions 5-7 are reserved for future protocol use.
+	// Bit positions 8-16 are available for blob-specific definitions.
+
+	// CRC options
+	REQ_CRC     = "REQ_CRC"
+	RES_CRC     = "RES_CRC"
+	NO_CRC      = "NO_CRC"
+	REQ_RES_CRC = "REQ_RES_CRC"
+
+	// OEM
+	OPENBMC = "OPENBMC"
+)
+
+// Maps OEM names to a 3 byte OEM number.
+// OENs are typically serialized as the first 3 bytes of a request body.
+var OENMap = map[string][3]uint8{
```

Comment:
If I'm understanding correctly, this is for the same purpose as the map here https://github.com/u-root/u-root/blob/master/pkg/ipmi/oem.go#L58

Does it make sense to combine the 2?

Reply:
seems like it to me, I'd +1

Reply:
Made the change!